subxt/src/constants/constants_client.rs

```diff
@@ -39,13 +39,14 @@ impl<T: Config, Client: OfflineClientT<T>> ConstantsClient<T, Client> {
             let expected_hash = self
                 .client
                 .metadata()
-                .constant_hash(address.pallet_name(), address.constant_name())?;
+                .pallet_by_name(address.pallet_name())
```

Comment:
dq: is this to ensure that the pallet name was stripped?

Reply:
I'm not sure I follow; perhaps you could elaborate?:)

Reply:
I was referring to changed API:

previous version:

```rust
 client.metadata().constant_hash(address.pallet_name(), address.constant_name())?;
```

new code:

```rust
   client
        .metadata()
        .pallet_by_name(address.pallet_name())
        .ok_or_else(|| MetadataError::PalletNameNotFound(address.pallet_name().to_owned()))?
        .constant_hash(address.constant_name())
        .ok_or_else(|| MetadataError::ConstantNameNotFound(address.constant_name().to_owned()))?;
```

Reply:
Aah, I had a couple of reasons for breaking the API into two here:

1. By getting the pallet first, we then have the information needed to find the relevant constant/storage/etc rather than needing hashmaps with both pallet and constant/sotrage/whatever keys in.
2. Instead of needing to provide `Result`s back to say whether it was the pallet or constant that couldn't be found, the metadata can just return `Option`s in a lot of cases, and leave it up to eg Subxt how to present the error, which felt cleaner design wise.
3. There may be other reasons to want to access a pallet by its name or whatever, and in some cases (eg in this PR) it's useful to have both the pallet and the specific event/whatever details to hand back anyway :)